rewrite-java/src/main/java/org/openrewrite/java/format/WrappingAndBracesVisitor.java

```diff
@@ -210,4 +187,51 @@ private List<J.Modifier> withNewline(List<J.Modifier> modifiers) {
         }
         return super.visit(tree, p);
     }
+
+    private List<J.Annotation> wrapAnnotations(List<J.Annotation> annotations, String whitespace, WrappingAndBracesStyle.Annotations annotationsStyle) {
+        return ListUtils.map(annotations, (index, ann) -> {
+            switch (annotationsStyle.getWrap()) {
+                case DoNotWrap:
+                    if (ann.getPrefix().getWhitespace().contains("\n")) {
+                        ann = ann.withPrefix(ann.getPrefix().withWhitespace(Space.SINGLE_SPACE.getWhitespace()));
+                    }
+                    break;
+                case WrapAlways:
+                    if (index == 0) {
+                        break;
+                    }
+                    ann = ann.withPrefix(ann.getPrefix().withWhitespace((whitespace.startsWith("\n") ? "" : "\n") + whitespace));
+                    break;
+            }
```

Comment:
```suggestion
            if (annotationsStyle.getWrap() == DoNotWrap && ann.getPrefix().getWhitespace().contains("\n")) {
                return ann.withPrefix(ann.getPrefix().withWhitespace(Space.SINGLE_SPACE.getWhitespace()));
            } else if (annotationsStyle.getWrap() == WrapAlways && index != 0) {
                return ann.withPrefix(ann.getPrefix().withWhitespace((whitespace.startsWith("\n") ? "" : "\n") + whitespace));
            }
```

If we ever can use Java 21+ with pattern matching, your current code would be better 😉.